packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -370,7 +373,44 @@ exports.sourceNodes = async (
           if (d.body.included) {
             dataArray.push(...d.body.included)
           }
-          if (d.body.links && d.body.links.next) {
+
+          // If JSON:API extras is configured to add the resource count, we can queue
+          // all API requests immediately.
+          if (d.body.meta?.count) {
+            // If we hadn't added urls yet
+            if (
+              d.body.meta.count > 50 &&
+              !typeRequestsQueued.has(d.body.data[0]?.type)
+            ) {
+              typeRequestsQueued.add(d.body.data[0]?.type)
+
+              // Get count of API requests
+              // We round down as we've already gotten the first page.
+              const requestsCount = Math.floor(d.body.meta.count / 50)
+              reporter.verbose(
+                `queueing ${requestsCount} API requests for type ${d.body.data[0].type} which has ${d.body.meta.count} entities.`
+              )
+              if (d.body.links.next?.href) {
+                const parsedUrl = url.parse(d.body.links.next?.href)
+                const parsedQueryString = querystring.parse(parsedUrl.query)
+                await Promise.all(
+                  _.range(requestsCount).map(pageOffset => {
+                    pageOffset += 1
+                    // Construct query string.
+                    const newQuerystring = {
+                      ...parsedQueryString,
+                      "page[offset]": pageOffset * 50,
+                    }
+                    const newUrl = url.format({
+                      ...parsedUrl,
+                      search: querystring.stringify(newQuerystring),
+                    })
+                    return getNext(newUrl)
+                  })
```

Comment:
should we think about limiting max requests in flight?

Reply:
the requests get put in a queue so that's already accounted for